packages/dashboard-frontend/src/store/Workspaces/selectors.ts

```diff
@@ -39,7 +40,9 @@ export const selectAllWorkspaces = createSelector(
   (cheWorkspacesState, devWorkspacesState) => {
     return [
       ...cheWorkspacesState.workspaces,
-      ...devWorkspacesState.workspaces,
+      // we were needed to add a filter because at first, we can get a new workspace object without metadata
+      // and then will get the update event with metadata.
+      ...(devWorkspacesState.workspaces.filter(workspace => isDevfileV2Metadata(workspace.metadata))),
```

Comment:
Why do we need to keep such objects in the store without being able to use them? Wouldn't it be better to skip them and wait for updated objects?

Reply:
I have removed it. I localized the problem: the metadata.labels could be undefined.